sim/src/main/cc/bridges/BridgeHarness.cc

```diff
@@ -0,0 +1,50 @@
+// See LICENSE for license details.
+
+#include "BridgeHarness.h"
+
+#include "bridges/blockdev.h"
+#include "bridges/bridge_driver.h"
+#include "bridges/uart.h"
+
+BridgeHarness::BridgeHarness(const std::vector<std::string> &args,
+                             simif_t *simif)
+    : simif_peek_poke_t(simif, PEEKPOKEBRIDGEMODULE_0_substruct_create),
+      simulation_t(args) {}
+
+BridgeHarness::~BridgeHarness() {}
+
+void BridgeHarness::add_bridge_driver(bridge_driver_t *bridge) {
+  bridges.emplace_back(bridge);
+}
+
+void BridgeHarness::simulation_init() {
+#include "constructor.h"
+  for (auto &bridge : bridges) {
+    bridge->init();
+  }
+}
+
+int BridgeHarness::simulation_run() {
+  // Reset the device.
+  poke(reset, 1);
+  step(1);
+  poke(reset, 0);
+  step(1);
+
+  // Tick until all requests are serviced.
+  step(get_step_limit(), false);
+  for (unsigned i = 0; i < get_tick_limit() && !simif->done(); ++i) {
```

Comment:
If you're going to provide a useful and generic simulation_run you should probably check the bridges don't signal for terminate, and treat it as an error if so. 

Also, hitting the tick limit likely implies deadlock, and should be treated as an error too, unless the watchdog is configured differently.

Reply:
This details are not particularly clear to me - we should discuss what a proper harness looks like offline and evolve it from this loop. Currently, if the tick limit is hit, then bridges will not have produced the correct output and the test will fail.

Reply:
In particular, in #1342 I am trying to unify all harnesses. We can get these tests to obey the timeout flag parsed by that class afterwards and display a consistent error message.